### PR TITLE
[REF] pos_restaurant: Split _get_order_lines method

### PR DESCRIPTION
**Pull request by CarmenMiranda**
Splitting the method `_get_order_lines` will allow making it inheritable, this way more order line fields can be added and expand them easily and avoid having to overwrite it.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by hugho-ad — APPROVED** [review]

**Comment by trj-odoo** [comment]
Hi @CarmenMiranda, could you explain how to reproduce the bug you are describing ?
Yeah sure splitting it make it easier to inherit but I don't get how that solve the issue you are mentioning.
`_get_fields_for_order_line` is where you add the field you want to load for the pos_restaurant

**Comment by CarmenMiranda** [comment]
Hi @trj-odoo I removed the comment about the issue as it was misleading about what is done in the PR, that was reported in the issue  https://github.com/odoo/odoo/issues/105604 and I think is being solved in another PR

Sorry for the confusion, by himself, this PR didn't solve the issue completely, this was more to allow the inheritance and in another one solve the issue like the method `_order_line_fields`

This PR was based on the PR https://github.com/odoo/odoo/pull/86067

File: addons/pos_restaurant/models/pos_order.py [hunk 1]
```diff
@@ -81,17 +100,7 @@ def _get_order_lines(self, orders):
...
-        extended_order_lines = []
-        for order_line in order_lines:
-            order_line['product_id'] = order_line['product_id'][0]
-            order_line['server_id'] = order_line['id']
-
-            del order_line['id']
-            if not 'pack_lot_ids' in order_line:
-                order_line['pack_lot_ids'] = []
-            else:
-                order_line['pack_lot_ids'] = [[0, 0, lot] for lot in order_line['pack_lot_ids']]
```
Comment (trj-odoo) [inline addons/pos_restaurant/models/pos_order.py:93]: It'll be better to just export this code to another function called (something like) `_prepare_order_lines(order_line)`  
and then do `extended_order_lines.append([0, 0, self._prepare_order_lines(order_line)`.

That way, you will never have to deal with the fact that it's a list of list (`[[0,0, order_line]] in the return during the override

**Comment by trj-odoo** [comment]
@robodoo r+

**Pull request closed without merging**